busy_beaver/adapters/github.py

```diff
@@ -50,8 +52,7 @@ def _get_all_items(self, url, *, max_pages=5) -> List[Dict]:
 
         all_items = []
         for page_num in range(1, min(last_page, max_pages) + 1):
-            combined_params = self.params.copy()
-            combined_params.update({"page": page_num})
+            combined_params = ChainMap({"page": page_num}, self.params)
```

Comment:
Might make sense to make `self.param` into a ChainMap and then update it as needed.